### PR TITLE
Kernel: Some more error propagation and String elimination

### PR DESCRIPTION
**Pull request by IdanHo**

### CODE REVIEW

File: Userland/Libraries/LibELF/Validation.cpp [hunk 1]
```diff
@@ -192,7 +192,7 @@ bool validate_elf_header(const ElfW(Ehdr) & elf_header, size_t file_size, bool v
     return true;
 }
 
-bool validate_program_headers(const ElfW(Ehdr) & elf_header, size_t file_size, const u8* buffer, size_t buffer_size, String* interpreter_path, bool verbose)
+bool validate_program_headers(const ElfW(Ehdr) & elf_header, size_t file_size, const u8* buffer, size_t buffer_size, char (*interpreter_path_buffer)[PATH_MAX], bool verbose)
```
Comment (IdanHo) [inline Userland/Libraries/LibELF/Validation.cpp]: I don't love this, but I can't think of a better way.

Comment (bgianfo) [reply Userland/Libraries/LibELF/Validation.cpp]: We could use a StringBuilder preallocated to PATH_MAX by the caller? That might be a tiny improvement?

Comment (IdanHo) [reply Userland/Libraries/LibELF/Validation.cpp]: Actually yeah, I hadn't considered that, we can just use a normal StringBuilder, and make it return ErrorOr<bool>, which will allow us to also handle this in Userland in the future.

Comment (IdanHo) [reply Userland/Libraries/LibELF/Validation.cpp]: Updated to use ErrorOr<bool> + StringBuilder*

**Pull request merged**